selfservice/strategy/code/strategy_recovery.go

```diff
@@ -394,19 +208,23 @@ func (s *Strategy) recoveryIssueSession(w http.ResponseWriter, r *http.Request,
 	}
 	sf.RequestURL, err = x.TakeOverReturnToParameter(f.RequestURL, sf.RequestURL, returnTo)
 	if err != nil {
-		return s.retryRecoveryFlowWithError(w, r, flow.TypeBrowser, err)
+		return s.retryRecoveryFlow(w, r, f.Type, RetryWithError(err))
 	}
 
 	config := s.deps.Config()
 
 	sf.UI.Messages.Set(text.NewRecoverySuccessful(time.Now().Add(config.SelfServiceFlowSettingsPrivilegedSessionMaxAge(ctx))))
 	if err := s.deps.SettingsFlowPersister().UpdateSettingsFlow(r.Context(), sf); err != nil {
-		return s.retryRecoveryFlowWithError(w, r, f.Type, err)
+		return s.retryRecoveryFlow(w, r, f.Type, RetryWithError(err))
 	}
 
-	if x.IsJSONRequest(r) {
+	switch {
+	case f.Type.IsAPI():
+		f.ContinueWith = append(f.ContinueWith, flow.NewContinueWithSettingsUI(sf))
```

Comment:
This would also be a breaking change right? Because the payload has changed and thus the client code needs to be adapted

Reply:
>I am guessing, there is actually no one hitting this code path with API flows in prod right now, because there is no way to continue with the settings flow after this.

> We've had a few questions around this, but AFAIR, we always ended on "please use the browser flows" instead, as there is (actually was, since you can tokenize a cookie to a token now) no way to get a session token from this response.

https://github.com/ory/kratos/pull/3273/files/058dd0104e34e9b50b2711b6f595bd6173da8f72#r1333047866

Technically, this is a breaking change, yes. But there is likely no one hitting this path right now. 

However, just to be safe, we should probably stick the new behavior behind a feature flag. Agreed.

Reply:
I put this behind the feature flag now, as discussed in the other conversations.